sansa-rdf-spark/src/main/scala/net/sansa_stack/rdf/spark/model/hdt/TripleOps.scala

```diff
@@ -0,0 +1,260 @@
+      spark.catalog.listTables()
+      logger.info(s"select ${TripleOps.SUBJECT_TABLE}.${TripleOps.INDEX_COL} as ${TripleOps.SUBJECT_COL}, ${TripleOps.PREDICATE_TABLE}.${TripleOps.INDEX_COL} as ${TripleOps.PREDICATE_COL},${TripleOps.OBJECT_TABLE}.${TripleOps.INDEX_COL} as ${TripleOps.OBJECT_COL} from ${TripleOps.TRIPLE_RAW_TABLE} join ${TripleOps.SUBJECT_TABLE} on ${TripleOps.TRIPLE_RAW_TABLE}.${TripleOps.SUBJECT_COL}=${TripleOps.SUBJECT_TABLE}.${TripleOps.NAME_COL} " +
+        s"join ${TripleOps.OBJECT_TABLE} on ${TripleOps.TRIPLE_RAW_TABLE}.${TripleOps.OBJECT_COL}=${TripleOps.OBJECT_TABLE}.${TripleOps.NAME_COL} " +
+        s"join ${TripleOps.PREDICATE_TABLE} on ${TripleOps.TRIPLE_RAW_TABLE}.${TripleOps.PREDICATE_COL}=${TripleOps.PREDICATE_TABLE}.${TripleOps.NAME_COL}")
+
+      //Creating Fact table from Subject,Predicate and Object index. Fact table contains unique ID of Subject/Object/Predicate
+      tripleFactTable= spark.sqlContext.sql(s"select ${TripleOps.SUBJECT_TABLE}.${TripleOps.INDEX_COL} as ${TripleOps.SUBJECT_COL}, ${TripleOps.PREDICATE_TABLE}.${TripleOps.INDEX_COL} as ${TripleOps.PREDICATE_COL},${TripleOps.OBJECT_TABLE}.${TripleOps.INDEX_COL} as ${TripleOps.OBJECT_COL} from ${TripleOps.TRIPLE_RAW_TABLE} join ${TripleOps.SUBJECT_TABLE} on ${TripleOps.TRIPLE_RAW_TABLE}.${TripleOps.SUBJECT_COL}=${TripleOps.SUBJECT_TABLE}.${TripleOps.NAME_COL} " +
+        s"join ${TripleOps.OBJECT_TABLE} on ${TripleOps.TRIPLE_RAW_TABLE}.${TripleOps.OBJECT_COL}=${TripleOps.OBJECT_TABLE}.${TripleOps.NAME_COL} " +
+        s"join ${TripleOps.PREDICATE_TABLE} on ${TripleOps.TRIPLE_RAW_TABLE}.${TripleOps.PREDICATE_COL}=${TripleOps.PREDICATE_TABLE}.${TripleOps.NAME_COL}");
+
+      logger.info(s"Registering Triples Fact Tables as ${TripleOps.TRIPLE_TABLE} " )
+      if(registerAsTable)
+        tripleFactTable.createOrReplaceTempView(TripleOps.TRIPLE_TABLE)
+    }
+    else if(compressedDir!=null && !compressedDir.isEmpty){
+
+      tripleFactTable=spark.read.schema(tripleSchema).csv(compressedDir+TripleOps.TRIPLE_DIR)
+      subjectDF=spark.read.schema(dictionarySchema).csv(compressedDir+TripleOps.SUBJECT_DIR)
+      objectDF=spark.read.schema(dictionarySchema).csv(compressedDir+TripleOps.OBJECT_DIR)
+      predicateDF=spark.read.schema(dictionarySchema).csv(compressedDir+TripleOps.PREDICATE_DIR)
+      if(registerAsTable)
+      {
+        registerDictionariesAsView(subjectDF,objectDF,predicateDF)
+        tripleFactTable.createOrReplaceTempView(TripleOps.TRIPLE_TABLE)
+      }
+
+    }
+    else{
+      logger.error("Input File or Compressed Directory is not provided. Please provide any one input.")
+    }
+    (tripleFactTable,subjectDF,objectDF,predicateDF)
+  }
+
+  /**
+    * Function saves the Index and Dictionaries Dataframes into given location
+    * @param outputDir Path to be written
+    * @param mode SaveMode of Write
+    */
+  def saveAsCSV(outputDir:String, mode:SaveMode): Unit =
```

Comment:
um. why do we need to save them as a csv? Aren't they just indexes? which could be kept in any file (parquet files, hadoop files?).

Reply:
That is the future enhancement to support various target types and compression type.